### PR TITLE
Consolidating Delayed Neutron Fraction Settings

### PR DESCRIPTION
**Pull request by jakehader**
Adding the functionality for the effective delayed neutron fraction, group-wise delayed neutron fraction, and group-wise precursor decay constants to be initialized on the `Core` using the `beta` and `decayConstants` settings. This is useful for supporting plugins that need access to delayed neutron data for reactor kinetics evaluations or representing absolute reactivity (dk/kk' or pcm) in dollars or cents (i.e., shutdown margin, control rod worth, excess reactivity, etc.)

### CODE REVIEW

File: armi/reactor/reactors.py [hunk 1]
```diff
@@ -223,6 +220,98 @@ def setOptionsFromCs(self, cs):
...
+
+            if isinstance(value, list):
+                return [float(val) for val in value]
+            elif isinstance(value, float):
+                return value
+            else:
+                raise ValueError(
+                    "`beta` must be set as a float or a list of floats. "
+                    f"Attempted to assign {value}."
+                )
```
Comment (youngmit) [inline armi/reactor/reactors.py:252]: This sort of check should be part of the Setting schema. By the time any code gets access to `cs["beta"]` it should already be in its canonical form. Is the voluptuous schema below not performing the conversion?

File: armi/reactor/reactors.py [hunk 2]
```diff
@@ -223,6 +220,98 @@ def setOptionsFromCs(self, cs):
...
+
+        def _decayConstantsValidator(value):
+            """Validate the input for precursor decay constants."""
+            if value is None:
+                return None
+
+            if isinstance(value, list):
+                return [float(val) for val in value]
```
Comment (youngmit) [inline armi/reactor/reactors.py:260]: Same as above

File: armi/reactor/reactors.py [hunk 3]
```diff
@@ -223,6 +220,98 @@ def setOptionsFromCs(self, cs):
...
+            )
+
+        # Report to the user the values were not applied.
+        if not reportTableData and (beta is not None or decayConstants is not None):
+            runLog.warning(
+                f"Delayed neutron fraction(s) - {beta} and decay constants - {decayConstants} have not been applied."
+            )
+            return
+
+        runLog.info(
```
Comment (youngmit) [inline armi/reactor/reactors.py:308]: I think an `else` would be more clear here than an early return

File: armi/reactor/tests/test_reactors.py [hunk 4]
```diff
@@ -200,8 +201,106 @@ def setUpClass(cls):
     def tearDownClass(cls):
         cls.directoryChanger.close()
 
+    def test_kineticsParameterAssignment(self):
```
Comment (youngmit) [inline armi/reactor/tests/test_reactors.py:204]: Great test

Comment (youngmit) [reply armi/reactor/tests/test_reactors.py:204]: Might be a bit expensive, though. Would it be possible to just make an empty Core object and pass it a bunch of different settings?

File: doc/user/inputs/settings.rst [hunk 5]
```diff
@@ -97,3 +97,39 @@ Detail assembly numbers
...
+In reactor physics analyses it is standard practice to represent reactivity
+in either absolute units (i.e., dk/kk' or pcm) or in dollars or cents. To
+support this functionality, the framework supplies the ``beta`` and
+``decayConstants`` settings to apply the delayed neutron fraction and
+precursor decay constants to the Core parameters during initialization.
+
+These settings come with a few caveats:
+
+    1. The ``beta`` setting supports two different meanings depending on
+       the type that is provided. If a float is given, then this setting
```
Comment (youngmit) [inline doc/user/inputs/settings.rst:114]: `float` is maybe a bit to programmery for user input docs. Maybe just "If a single value is provided..."

File: doc/user/inputs/settings.rst [hunk 6]
```diff
@@ -97,3 +97,39 @@ Detail assembly numbers
...
+       is interpreted as the effective delayed neutron fraction for the
+       system and is applied to the Core's ``beta`` parameter. If a list
```
Comment (youngmit) [inline doc/user/inputs/settings.rst:116]: Discussing the ``beta`` parameter directly might be TMI in this context.

File: armi/reactor/tests/test_reactors.py (hunk 4 above)
Comment (jakehader) [reply armi/reactor/tests/test_reactors.py:204]: Sure thing - should be an easy swap.

File: armi/reactor/reactors.py (hunk 1 above)
Comment (jakehader) [reply armi/reactor/reactors.py:252]: Yeah I think I put the voluptuous checks in after the fact, so I might just be able to delete this now. I'll try it.

File: armi/reactor/reactors.py (hunk 3 above)
Comment (jakehader) [reply armi/reactor/reactors.py:308]: Sounds good.

File: doc/user/inputs/settings.rst (hunk 6 above)
Comment (jakehader) [reply doc/user/inputs/settings.rst:116]: That's true, but I want to make it clear how the setting is being applied versus just saying what it means. Is this just confusing to a new reader or is there a better place to put this type of information? I guess I could have this in a docstring too and then link directly to it.

File: doc/user/inputs/settings.rst (hunk 5 above)
Comment (jakehader) [reply doc/user/inputs/settings.rst:114]: Gottcha

File: armi/reactor/tests/test_reactors.py (hunk 4 above)
Comment (jakehader) [reply armi/reactor/tests/test_reactors.py:204]: Yeah that is easy enough. The test is ~0.5 seconds without the change though.

Comment (jakehader) [reply armi/reactor/tests/test_reactors.py:204]: Moving to an empty reactor sped the test up by ~40%

File: armi/reactor/reactors.py (hunk 1 above)
Comment (jakehader) [reply armi/reactor/reactors.py:252]: Yeah this was just redundant with voluptuous so it's gone now.

File: armi/reactor/reactors.py [hunk 7]
```diff
@@ -223,6 +220,58 @@ def setOptionsFromCs(self, cs):
...
         self._minMeshSizeRatio = cs["minMeshSizeRatio"]
+        self._setupEffectiveDelayedNeutronFraction(cs)
+
+    def _setupEffectiveDelayedNeutronFraction(self, cs):
+        """Process the settings for the delayed neutron fraction and precursor decay constants."""
+        # Verify and set the core beta parameters based on the user-supplied settings
+        runLog.info(
+            "Checking for user-supplied `beta` and `decayConstants` settings and attempting to "
+            f"apply them to state of {self}."
+        )
```
Comment (ntouran) [inline armi/reactor/reactors.py:231]: info is too verbose for this level

File: armi/reactor/reactors.py [hunk 8]
```diff
@@ -223,6 +220,58 @@ def setOptionsFromCs(self, cs):
         self._circularRingPitch = cs["circularRingPitch"]
         self._automaticVariableMesh = cs["automaticVariableMesh"]
         self._minMeshSizeRatio = cs["minMeshSizeRatio"]
+        self._setupEffectiveDelayedNeutronFraction(cs)
+
+    def _setupEffectiveDelayedNeutronFraction(self, cs):
```
Comment (ntouran) [inline armi/reactor/reactors.py:225]: probably belongs on framework neutronics plugin

File: armi/reactor/reactors.py [hunk 9]
```diff
@@ -223,6 +220,58 @@ def setOptionsFromCs(self, cs):
...
+                ("Group-wise Precursor Decay Constants", self.p.betaDecayConstants)
+            )
+
+        # Report to the user the values were not applied.
+        if not reportTableData and (beta is not None or decayConstants is not None):
+            runLog.warning(
+                f"Delayed neutron fraction(s) - {beta} and decay constants - {decayConstants} have not been applied."
+            )
+        else:
+            runLog.info(
```
Comment (ntouran) [inline armi/reactor/reactors.py:268]: extra

**Pull request merged**